src/databricks/labs/ucx/source_code/s3fs.py

```diff
@@ -0,0 +1,62 @@
+import ast
+from collections.abc import Iterable
+
+from databricks.labs.ucx.source_code.base import Advice, Linter, Deprecation
+
+
+class DetectS3fsVisitor(ast.NodeVisitor):
+    def __init__(self):
+        self._advices: list[Advice] = []
+
+    def visit_Import(self, node):  # pylint: disable=invalid-name
+        for alias in node.names:
+            if alias.name == 's3fs':
```

Comment:
We have DependencyGraph object with dependencies method, that returns file/cell dependencies, including imports. 

Rewrite this to fit into that framework. There'll be more incompatible packages/subpackages we'll have to flag.

The good part is that you already have unit tests for this.

Reply:
No problem.

Reply:
I will do this in two separate commits. One to use the whitelist to detect s3fs and raise Advice, One will modify the processing so we can identify where the the import came for as this information is currently lost